arch/x86/sysctx.c

```diff
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+#include <uk/assert.h>
+#include <uk/arch/ctx.h>
+
+__uptr ukarch_sysctx_get_tlsp(struct ukarch_sysctx *sysctx)
+{
+	UK_ASSERT(sysctx);
+
+	return sysctx->fs_base;
+}
+
+void ukarch_sysctx_set_tlsp(struct ukarch_sysctx *sysctx, __uptr tlsp)
+{
+	UK_ASSERT(sysctx);
+
+	uk_pr_debug("System call updated userland TLS pointer register to %p (before: %p)\n",
```

Comment:
I find the debug message misleading by just seeing the library functions here. I know it is mostly called from system calls but the API allows to call it also from other places. I'd either remove the message or generalize it more (not mention system call). The other question that pops up if we should put then a message to gs_base, as well.